src/puppet.js

```diff
@@ -89,7 +89,142 @@
 
   }
 
-  function PuppetNetworkChannel(puppet, remoteUrl, useWebSocket, onReceive, onSend, onError, onStateChange) {
+  /**
+   * @constructor
+     */
+  function Reconnector(eventDispatcher, reconnectAction) {
+    this._eventDispatcher = eventDispatcher;
+    this._reconnect = reconnectAction;
```

Comment:
I see no reason to change them neither. It's a lot harder to remove something from public API then to add it, so I would leave it private. If need occurs we can always add it later